trainer/derrian_backend/sd_scripts/sdxl_train.py

```
@@ -618,11 +870,159 @@ def optimizer_hook(parameter: torch.Tensor):
+    loss_recorder = train_util.EMARecorder()
+    val_loss_recorder = train_util.EMARecorder()
+
+    if args.edm2_loss_weighting:
+        loss_scaled_recorder = train_util.EMARecorder()
+        loss_edm2_recorder = train_util.EMARecorder()
+
+    def calculate_val_loss(epoch_step):
+        if not train_util.calculate_val_loss_check(args, global_step, epoch_step, val_dataloader, train_dataloader):
+            return None, None, {}
+
+        rng_states = switch_rng_state(int(args.validation_seed) if args.validation_seed else 23, accelerator)
```

Ruff (0.15.12)

[error] Undefined name `accelerator`

(F821)